staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3057,11 +3057,19 @@ type PodStatus struct {
 	// IP address of the host to which the pod is assigned. Empty if not yet scheduled.
 	// +optional
 	HostIP string `json:"hostIP,omitempty" protobuf:"bytes,5,opt,name=hostIP"`
-	// IP address allocated to the pod. Routable at least within the cluster.
+	// Default IP address allocated to the pod. Routable at least within the cluster.
 	// Empty if not yet allocated.
 	// +optional
 	PodIP string `json:"podIP,omitempty" protobuf:"bytes,6,opt,name=podIP"`
 
+	// IP addresses allocated to the pod with associated metadata.
+	// This list is inclusive, i.e. it includes the default IP stored in the
+	// "PodIP" field. It is empty if no IPs have been allocated yet.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	PodIPs []PodIPInfo `json:"podIPs,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,12,rep,name=podIPs"`
```

Comment:
I think we should get a PR that *just* adds this new field with very clear semantics and tests.

Reply:
@thockin - There's a WIP PR for adding the plural PodIPs: PR #69153 . There's a little bit of unavoidable overlap between these 2 PRs. PR #69153 adds PodIPs[0] references where PodIP used to be used.

Reply:
@leblancd  @thockin I'll rebase on that  PR #69153 after it's merged